1-labeled-tuples/1-labeled-tuples.md

```diff
@@ -0,0 +1,258 @@
+<!-- TODO: what do we have now -->
+Currently only unlabeled tuples are supported in Hazel, and elements can only be accessed positionally using pattern matching (via let or case).
+# Labeled Product Types
+Syntax: `.label1 ty1, .label2 ty2, ..., .labeln tyn`
+
+Files to Edit: UHTyp.re
+<!-- TODO: add `.label` as a new type form <br/> -->
+<!-- TODO: List the files you expect ot edit for each thing, and a sentance about what you expect to do -->
+<!-- Just add space operator, check UHExp.re for this -->
+<!-- Look at Skeltype parser and see that there is something that determines prescedence,-->
+<!-- Stat that we are not requiring parenthesis-->
+
+To add labeled product types, the operator and opearnd types will need to be expanded like so:
+```
+type operator =
+...
+ | Space; 
+```
+<!-- For space, say if left or right associative and give presedence-->
+Space will be left associateive and have the highest precedence. 
+```
+type operand =
+...
+ | Label(Label.t);
+```
+
+<!-- TODO: do we want to allow partially labeled product types? -->
+Partially labeled product types are allowed, and labels and non-labeled positions can be interleaved.
+For example, the labeled product type `(.x Num, Num, .y Num)` is allowed. 
+
+Singleton label products are supported. For example, `.x Num` is supported.
+
+### Type Equivalence
+
+A labeled product may still be used with positional arguments, so the type equivalence considers the order of the labels with equlivalence. For example, `(.x Num, .y Num, .z Num) != (.z Num, .y Num, .x Num)` because the order of the labels is different.
+
+### Type Syntax Errors
+Files to Edit: CursorInfo.re
+
+* A label must be followed by a valid type and comma operator, not another label. For example, `.label1 .label2 ty` produces an error.
+ * Error Cursor appears on `.label2`
+ * Expecting a Type Got a Label
+
+* A label cannot exist by itself; it is given meaning by having a type follow it. For example, `.label1 ` by itself produces an error. 
+ * Error Cursor appears on `.label1`
+ * Message: Expecting a Type Got a Label
+
+* Elements in the tuples need to be separated by commas, if they are not then this produces an error. For example, `.label1 ty .label2` produces an error. 
+ * Error cursor appears on space operator
+ * Message: Expecting ? Got an Unexpected Label
+
+* Duplicate labels within a tuple are not valid, so they produce an error. The error will appear on the subsequent duplicate uses, not on the type as a whole. FOr example, the type `.label1 Num, .label1 Num, .label2 Num, .label1 Bool` will have errors on the second use of `.label1 Num` and `.label1 Bool`
+ * Error Cursor appears on second and third use of `.label1`<br/>
+ * Expecting a Unique Label Got a Duplicate Label
+
+# Labeled Tuple Expressions

```

Comment:
Expressions as the parent section, then "labeled tuples" and "projection" are subsections